pkg/argocd/applications.go

```diff
@@ -202,3 +202,44 @@ func (builder *ApplicationBuilder) validate() (bool, error) {
 
 	return true, nil
 }
+
+// WithGitDetails applies git details to application definition.
+func (builder *ApplicationBuilder) WithGitDetails(gitRepo, gitBranch, gitPath string) *ApplicationBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if gitRepo == "" {
+		glog.V(100).Infof("The 'gitRepo' of the argocd application is empty")
+
+		builder.errorMsg = "'gitRepo' parameter is empty"
+	}
+
+	if gitBranch == "" {
+		glog.V(100).Infof("The 'gitBranch' of the argocd application is empty")
+
+		builder.errorMsg = "'gitBranch' parameter is empty"
+	}
+
+	if gitPath == "" {
+		glog.V(100).Infof("The 'gitPath' of the argocd application is empty")
+
+		builder.errorMsg = "'gitPath' parameter is empty"
+	}
+
+	glog.V(100).Infof(
+		"Redefining argocd application: %s in namespace: %s with the following git details "+
```

Comment:
You're not neccessarilty redefining the application, so I think it's better to change the wording to the following
```suggestion
		"Adding the following git details to the argocd application: %s in namespace: %s "+
```